Framework/Algorithms/src/PDFFourierTransform.cpp

```diff
@@ -229,12 +231,12 @@ double PDFFourierTransform::determineRho0() {
 void PDFFourierTransform::exec() {
   // get input data
   API::MatrixWorkspace_const_sptr inputWS = getProperty("InputWorkspace");
-  MantidVec inputQ = inputWS->dataX(0);                   //  x for input
+  auto inputQ = inputWS->x(0).rawData();                    //  x for input
   HistogramData::HistogramDx inputDQ(inputQ.size(), 0.0); // dx for input
   if (inputWS->sharedDx(0))
     inputDQ = inputWS->dx(0);
-  MantidVec inputFOfQ = inputWS->dataY(0);  //  y for input
-  MantidVec inputDfOfQ = inputWS->dataE(0); // dy for input
+  auto inputFOfQ = inputWS->y(0).rawData();  //  y for input
+  auto inputDfOfQ = inputWS->e(0).rawData(); // dy for input
```

Comment:
@PranavBahuguna Ok, I tried on my system. Results with you current version of the branch:

```
time="0.117505"
time="0.118494"
time="0.117985"
time="0.117923"
time="0.118178"
time="0.118628"
time="0.117796"
time="0.118586"
time="0.118522"
time="0.118005"
time="0.117125"
time="0.118476"
time="0.118438"
time="0.116519"
time="0.091305"
time="0.116158"
time="0.117869"
time="0.117811"
time="0.118388"
time="0.117609"
time="0.118731"
time="0.118337"
time="0.118284"
time="0.118379"
time="0.093599"
time="0.117647"
time="0.119212"
time="0.116286"
time="0.118516"
time="0.11828"
time="0.118359"
time="0.117296"
time="0.118426"
time="0.118512"
time="0.117882"
time="0.117709"
time="0.117535"
time="0.118007"
time="0.11779"
time="0.118151"
time="0.118473"
time="0.118582"
```

Results without `rawData` (and changing the function parameters accordingly to `const HistogramX &` and `const HistogramY &`):

```
time="0.117871"
time="0.118317"
time="0.118508"
time="0.117998"
time="0.120211"
time="0.117685"
time="0.118358"
time="0.118707"
time="0.117841"
time="0.11694"
time="0.11804"
time="0.118446"
time="0.118019"
time="0.117642"
time="0.117816"
time="0.11837"
time="0.118566"
time="0.118079"
time="0.118693"
time="0.118461"
time="0.117614"
time="0.11872"
time="0.11746"
time="0.11896"
time="0.11758"
time="0.117863"
time="0.116313"
time="0.117515"
time="0.11747"
time="0.117942"
time="0.116959"
time="0.118845"
```

This is with gcc on Linux. I can't see any difference. What system are you running on? Could you try again?